### DISCUSSION TITLE
Control denoise step - when is it applied?

### DISCUSSION THREAD

**Discussion by lbeltrame** [vladmandic/sdnext#2745] [Q&A]
Hello,

I switched to dev and tried Control, and I'm having a blast with it: being able to do poses with ControlNet and SDXL is absolutely awesome! (I also have a rough idea on how to use the pipeline, since I saw some struggled with it; poke if you want some details). 

I'm trying to recreate a "second pass" workflow using the resize operations in Control, but what I'm not sure about is when the denoising is applied (as I didn't see the same steps in the terminal as when doing traditional txt2img and second pass). As far as I can tell, it is applied on the *first* step, that is when the control image is resized prior to actually generating the image, rather than on the second resizing step. 

Is my interpretation correct, or am I getting things wrong?

**Comment by vladmandic** [comment]
its a global setting, it should be applied to both.
steps do not change with denoising strength as denoising strength is applied as strength in diffusers pipeline.

**Reply by lbeltrame** [reply]
Is there a 1:1 overlap between the denoise strength from the regular txt2img workflow and Control? I made a couple tests and the images seemed to lose a bit of details (*entirely* unscientific and subjective), but I guess that due to the conditioning by ControlNet you need to crank denoise a little higher.

**Reply by vladmandic** [reply]
its very similar, but entire module is brand new and not designed really to behave "identically" - i'm trying new things. denoising should be 99% similar, masking on the other hand is 99% different.

**Reply by lbeltrame** [reply]
I usually notice if something is amiss: but in the worst case, nothing prevents me from feeding the image back to Control and run img2img to increase details (keeping the ControlNet constraints). 
That said I only ran a few tests, I hope to do more in the next few days and in the weekend.

**Comment by vladmandic** [comment]
btw, since you're getting familiar with workflows - i'd love if you touched base with @Aptronymist as he's preparing the docs for control module!

**Reply by lbeltrame** [reply]
I'd be happy to if I can help.

**Reply by Aptronymist** [reply]
Gotcha, and yeah, I'm working up the docs right now in fact for masking, which is why I didn't notice this, but I'd love to hear what you've worked up, as really I've never been a controlnet user for the most part, so I'm not the best person to cover its workflows with my limited experience.  

However you wish to collaborate is good by me, but discord is my preference.

**Reply by lbeltrame** [reply]
I would avoid Discord if possible as I don't like their privacy policy (I have one account which I only use to log in to CivitAI and nothing else). Any other useful means of collaboration you might like? Of course if nothing fits the bill I can use Discord at a last resort.

That said, I can provide examples of the workflows I used with the extension, and how I figured out how to use it here (SDXL doesn't have all the models SD has, but the concepts are very similar).

**Reply by Aptronymist** [reply]
Well something more or less back and forth chat-style in real-time I guess would be best, you can choose.
I'd be happy to see your workflows of course, anything you have tinkered with will save me time and provide insight, maybe I can document your "tricks" and the like, as I likely won't know them!

**Reply by lbeltrame** [reply]
Aside the aforementioned Discord account, I'm on Matrix: `@einar:conference.heavensinferno.net`.

**Reply by Aptronymist** [reply]
Okay, whatever that is I'm sure I can find it. It might be tomorrow though, kinda slammed at the moment and dinner is coming up soon.

**Reply by lbeltrame** [reply]
No problem. Easiest is to jump to https://app.element.io/ and use it from there. Start a "direct chat", and when asked for the contact name paste that thing I posted here (`@einar:conference.heavensinferno.net`), and done. If it doesn't work, we can go back to Discord.

**Reply by lbeltrame** [reply]
@Aptronymist Let me know when/if you have time, I'm about to sketch down some notes given what Vlad said in the discussion I opened + responses to issues.

**Reply by Aptronymist** [reply]
> @Aptronymist Let me know when/if you have time, I'm about to sketch down some notes given what Vlad said in the discussion I opened + responses to issues.

Hell, yeah, sorry. Things are a little out of hand this week/month. 
If you wouldn't mind, please use discord and join our server, after all, they can only steal/sell what you actually say, and I doubt our conversation would be of any value to anyone else.
That being said I'm eager to see what you are doing.

**Reply by vladmandic** [reply]
also, if you don't like or trust the discord app, you can just use the web client.

**Reply by lbeltrame** [reply]
No, it's the privacy policy I don't trust. Well, I'll make a throwaway account then.

**Reply by vladmandic** [reply]
oh, that's totally reasonable
